File: src/main/java/team6458/util/CameraSetup.java
package team6458.util;

import edu.wpi.first.wpilibj.CameraServer;

import java.util.logging.Level;
import java.util.logging.Logger;

/**
 * A utility class to start USB camera capture data.
 */
public final class CameraSetup {

    public static final String DEFAULT_CAMERA_NAME = "cam0";
    public static final int DEFAULT_CAMERA_ID = 0;
    private static final Logger logger = Logger.getLogger(CameraSetup.class.getName());

    /**
     * No instantiation.
     */
    private CameraSetup() {
    }

    /**
     * Sets up the default camera.
     *
     * @see #setupCamera
     * @see #DEFAULT_CAMERA_NAME
     * @see #DEFAULT_CAMERA_ID
     */
    public static synchronized boolean setupDefaultCamera() {
        return setupCamera(DEFAULT_CAMERA_NAME, DEFAULT_CAMERA_ID);
    }

    /**
     * Sets up automatic capture for a camera with a given name and ID.
     *
     * @param name The name for the camera
     * @param id   The integral ID for the camera
     * @return True if successful, false otherwise
     */
    public static synchronized boolean setupCamera(String name, int id) {
        try {
            CameraServer.getInstance().startAutomaticCapture(name, id);
            return true;
        } catch (Throwable t) {
            t.printStackTrace();
        }

        logger.log(Level.WARNING, "Failed to start camera capture (" + name + ", " + id + ")");

        return false;
    }

}

File: src/main/kotlin/team6458/util/CameraSetup.kt
package team6458.util

import edu.wpi.first.wpilibj.CameraServer
import team6458.util.CameraSetup.setupDefaultCamera
import java.util.logging.Level
import java.util.logging.Logger


/**
 * Sets up cameras for the Driver Station. Use [setupDefaultCamera] for a camera with name `cam0` and ID `0`.
 */
object CameraSetup {

    private val logger = Logger.getLogger(this::class.java.name)

    private const val DEFAULT_CAMERA_NAME: String = "cam0"
    private const val DEFAULT_CAMERA_ID: Int = 0

    /**
     * Sets up a camera with the name [DEFAULT_CAMERA_NAME] and ID [DEFAULT_CAMERA_ID].
     *
     * @see setupCamera
     */
    fun setupDefaultCamera() = setupCamera(DEFAULT_CAMERA_NAME, DEFAULT_CAMERA_ID)

    /**
     * Sets up a camera with the [name] and [id].
     *
     * @return True if successful, false otherwise
     */
    fun setupCamera(name: String, id: Int): Boolean {
        try {
            CameraServer.getInstance()?.startAutomaticCapture(name, id) ?: error("CameraServer instance is null")
            return true
        } catch (t: Throwable) {
            t.printStackTrace()
        }

        logger.log(Level.WARNING, "Failed to initialize camera set-up!")

        return false
    }

}
